File: android/src/main/kotlin/id/co/rfbnr/flutter_indocard_ocr/ModuleOCR/model/KTPModel.kt
package id.co.rfbnr.flutter_indocard_ocr.ModuleOCR.model

import org.json.JSONObject

/**
 * KTP Data Model
 */
data class KTPModel(
    var nik: String? = null,
    var nama: String? = null,
    var tempatLahir: String? = null,
    var tanggalLahir: String? = null,
    var jenisKelamin: String? = null,
    var alamat: String? = null,
    var rtrw: String? = null,
    var kelurahan: String? = null,
    var kecamatan: String? = null,
    var agama: String? = null,
    var statusPerkawinan: String? = null,
    var pekerjaan: String? = null,
    var kewarganegaraan: String? = "WNI",
    var provinsi: String? = null,
    var kota: String? = null,
    var berlakuHingga: String? = "SEUMUR HIDUP",
    var confidence: Int = 0
) {
    /**
     * Convert to JSON Map
     */
    fun toJson(): Map<String, String> {
        return mapOf(
            "nik" to (nik ?: ""),
            "nama" to (nama ?: ""),
            "tempatLahir" to (tempatLahir ?: "-"),
            "tanggalLahir" to (tanggalLahir ?: "-"),
            "jenisKelamin" to (jenisKelamin ?: "-"),
            "alamat" to (alamat ?: "-"),
            "rtrw" to (rtrw ?: "000/000"),
            "kelurahan" to (kelurahan ?: "-"),
            "kecamatan" to (kecamatan ?: "-"),
            "agama" to (agama ?: "-"),
            "statusPerkawinan" to (statusPerkawinan ?: "-"),
            "pekerjaan" to (pekerjaan ?: "-"),
            "kewarganegaraan" to (kewarganegaraan ?: "WNI"),
            "provinsi" to (provinsi ?: "-"),
            "kota" to (kota ?: "-"),
            "berlakuHingga" to (berlakuHingga ?: "SEUMUR HIDUP")
        )
    }

    /**
     * Convert to JSON String
     */
    fun toJsonString(): String {
        val dict = toJson()
        val json = JSONObject(dict)
        return json.toString()
    }
}

File: ios/Classes/ModuleOCR/models/KTPModel.swift
import Foundation

struct KTPModel {
    var nik: String?
    var nama: String?
    var tempatLahir: String?
    var tanggalLahir: String?
    var jenisKelamin: String?
    var alamat: String?
    var rtrw: String?
    var kelurahan: String?
    var kecamatan: String?
    var agama: String?
    var statusPerkawinan: String?
    var pekerjaan: String?
    var kewarganegaraan: String?
    var provinsi: String?
    var kota: String?
    var berlakuHingga: String?
    var confidence: Int = 0

    init() {
        self.nik = nil
        self.nama = nil
        self.tempatLahir = nil
        self.tanggalLahir = nil
        self.jenisKelamin = nil
        self.alamat = nil
        self.rtrw = nil
        self.kelurahan = nil
        self.kecamatan = nil
        self.agama = nil
        self.statusPerkawinan = nil
        self.pekerjaan = nil
        self.kewarganegaraan = "WNI"
        self.provinsi = nil
        self.kota = nil
        self.berlakuHingga = "SEUMUR HIDUP"
        self.confidence = 0
    }

    // Convert to JSON dictionary
    func toJson() -> [String: String] {
        return [
            "nik": nik ?? "unknown",
            "nama": nama ?? "unknown",
            "tempatLahir": tempatLahir ?? "unknown",
            "tanggalLahir": tanggalLahir ?? "unknown",
            "jenisKelamin": jenisKelamin ?? "unknown",
            "alamat": alamat ?? "unknown",
            "rtrw": rtrw ?? "000/000",
            "kelurahan": kelurahan ?? "unknown",
            "kecamatan": kecamatan ?? "unknown",
            "agama": agama ?? "unknown",
            "statusPerkawinan": statusPerkawinan ?? "unknown",
            "pekerjaan": pekerjaan ?? "unknown",
            "kewarganegaraan": kewarganegaraan ?? "WNI",
            "provinsi": provinsi ?? "unknown-",
            "kota": kota ?? "unknown",
            "berlakuHingga": berlakuHingga ?? "SEUMUR HIDUP"
        ]
    }

    // Convert to JSON string
    func toJsonString() -> String? {
        let dict = toJson()
        guard let jsonData = try? JSONSerialization.data(withJSONObject: dict, options: []),
              let jsonString = String(data: jsonData, encoding: .utf8) else {
            return nil
        }
        return jsonString
    }
}
